# Stuff+ Analysis

In [3]:
data = pd.read_csv('../data/xwhiff_grouped_21_22.csv', index_col = [0])
data2 = pd.read_csv('../data/stuff_table_21_22.csv', index_col = [0])
data = data.round(1)
print(data.shape)
data.head()

(1367128, 14)


player_name  year p_throws pitch_type  velo  pfx_x  pfx_z  \
0  Santana, Dennis  2022        R         FF  96.9   12.5   14.5   
1  Santana, Dennis  2022        R         FF  97.5    9.7   13.2   
2  Santana, Dennis  2022        R         FF  98.2   11.6   14.4   
3  Santana, Dennis  2022        R         FF  96.8   12.5   13.9   
4  Santana, Dennis  2022        R         FF  98.4   10.1   16.2   

   release_pos_x  release_pos_z  release_extension  whiff  xWhiff  n_pitches  \
0            2.1            5.8                6.1      0     4.6          1   
1            2.2            5.7                6.0      1    68.4          1   
2            2.3            5.8                5.9      0    11.0          1   
3            2.2            5.7                6.0      0    11.0          1   
4            2.1            5.9                6.0      0     7.0          1   

   Stuff+  
0    48.2  
1   716.1  
2   115.2  
3   115.2  
4    73.3

In [4]:
print(data2.shape)
data2.head()

(3508, 12)


player_name  year p_throws  n_pitches pitch_type  velo  pfx_x  pfx_z  \
0   Abbott, Cory  2021        R        207         FF  92.6    9.6   17.2   
1   Abbott, Cory  2022        R        372         FF  91.5   11.6   16.9   
2  Abreu, Albert  2021        R        103         FF  97.8   11.7   15.4   
3  Abreu, Albert  2021        R        212         SI  98.0   15.3   13.3   
5  Abreu, Albert  2022        R        322         SI  98.6   14.7   13.1   

   release_pos_x  release_pos_z  release_extension  Stuff+  
0            2.6            5.5                6.2    91.4  
1            2.7            5.5                6.3    89.0  
2            2.5            5.7                6.4   123.3  
3            2.5            5.6                6.5    82.5  
5            1.7            5.7                6.6    75.9

In [ ]:
grouped_fb_r = rhp_fb.groupby(['player_name', 'p_throws', 'pitch_type'], 
                              as_index = False)['n_pitches'].sum()

avg_fb_r = rhp_fb.groupby(['player_name', 'p_throws', 'pitch_type'], 
                          as_index = False)['velo', 'pfx_x', 'pfx_z', 'release_pos_x', 'release_pos_z', 
                                            'release_extension', 'Stuff+'].mean()

In [14]:
stuff1 = data.groupby(['year', 'pitch_type'], as_index = False)['n_pitches'].sum()
stuff2 = data.groupby(['year', 'pitch_type'], as_index = False)['Stuff+'].mean()
stuff = pd.merge(stuff1, stuff2, how = 'left', on = ['year', 'pitch_type'])
stuff = stuff.round()
#stuff.head(10)
#print(stuff.to_markdown())

|    |   year | pitch_type   |   n_pitches |   Stuff+ |
|---:|-------:|:-------------|------------:|---------:|
|  0 |   2021 | FF           |      249680 |      111 |
|  1 |   2022 | FF           |      226564 |      110 |
|  2 |   2021 | SI           |      109150 |       79 |
|  3 |   2022 | SI           |      106077 |       75 |
|  4 |   2021 | FC           |       47444 |       84 |
|  5 |   2022 | FC           |       48924 |       82 |
|  6 |   2021 | SL           |      135534 |      113 |
|  7 |   2022 | SL           |      124097 |      111 |
|  8 |   2021 | CU           |       58770 |       89 |
|  9 |   2022 | CU           |       53589 |       86 |
| 10 |   2021 | KC           |       13672 |       98 |
| 11 |   2022 | KC           |       14647 |       97 |
| 12 |   2021 | CH           |       80338 |       97 |
| 13 |   2022 | CH           |       77339 |       98 |
| 14 |   2021 | FS           |       10842 |      118 |
| 15 |   2022 | FS           |       10461 |      117 |

In [15]:
stuff1 = data.groupby(['year', 'p_throws', 'pitch_type'], as_index = False)['n_pitches'].sum()
stuff2 = data.groupby(['year', 'p_throws', 'pitch_type'], as_index = False)['Stuff+'].mean()
stuff = pd.merge(stuff1, stuff2, how = 'left', on = ['year', 'p_throws', 'pitch_type'])
stuff = stuff.round()
#stuff.head(10)
#print(stuff.to_markdown())

|    |   year | p_throws   | pitch_type   |   n_pitches |   Stuff+ |
|---:|-------:|:-----------|:-------------|------------:|---------:|
|  0 |   2021 | R          | FF           |      176386 |      112 |
|  1 |   2022 | R          | FF           |      162398 |      111 |
|  2 |   2021 | L          | FF           |       73294 |      108 |
|  3 |   2022 | L          | FF           |       64166 |      109 |
|  4 |   2021 | R          | SI           |       75680 |       76 |
|  5 |   2022 | R          | SI           |       76701 |       73 |
|  6 |   2021 | L          | SI           |       33470 |       83 |
|  7 |   2022 | L          | SI           |       29376 |       79 |
|  8 |   2021 | R          | FC           |       31182 |       88 |
|  9 |   2022 | R          | FC           |       37078 |       82 |
| 10 |   2021 | L          | FC           |       16262 |       77 |
| 11 |   2022 | L          | FC           |       11846 |       82 |
| 12 |   2021 | R          | SL           |      100754 |      112 |
| 13 |   2022 | R          | SL           |       93351 |      110 |
| 14 |   2021 | L          | SL           |       34780 |      116 |
| 15 |   2022 | L          | SL           |       30746 |      114 |
| 16 |   2021 | R          | CU           |       39992 |       87 |
| 17 |   2022 | R          | CU           |       37386 |       86 |
| 18 |   2021 | L          | CU           |       18778 |       92 |
| 19 |   2022 | L          | CU           |       16203 |       88 |
| 20 |   2021 | R          | KC           |       11633 |      101 |
| 21 |   2022 | R          | KC           |       11996 |       98 |
| 22 |   2021 | L          | KC           |        2039 |       84 |
| 23 |   2022 | L          | KC           |        2651 |       92 |
| 24 |   2021 | R          | CH           |       50661 |       98 |
| 25 |   2022 | R          | CH           |       50374 |       95 |
| 26 |   2021 | L          | CH           |       29677 |       97 |
| 27 |   2022 | L          | CH           |       26965 |      103 |
| 28 |   2021 | R          | FS           |       10239 |      118 |
| 29 |   2022 | R          | FS           |       10106 |      118 |
| 30 |   2021 | L          | FS           |         603 |      120 |
| 31 |   2022 | L          | FS           |         355 |      103 |